indico/modules/events/management/forms.py

```diff
@@ -232,6 +232,12 @@ def __init__(self, *args, **kwargs):
             del self.label
             del self.label_message
 
+    def validate_keywords(self, field):
+        allowed_keywords = set(event_settings.get('allowed_keywords'))
```

Comment:
```suggestion
        allowed_keywords = set(event_settings.get('allowed_keywords')) | set(field.object_data)
```

That way you are not required to remove no-longer-valid keywords that are already there.

Reply:
I don't get this at all, sorry. If we allow any keyword to be stored we can simply remove the `validate_keyword` function altogether. Is that what you mean?

Reply:
`object_data` is the existing data that's already stored. Adding this ensures that existing data can stay even if it has been removed from the global whitelist.

Reply:
Oh! I see, I didn't read `object_data`, but just `data`. It makes sense now 😄